backend/danswer/server/manage/administrative.py

```diff
@@ -231,3 +235,56 @@ def update_token_budget_settings(
     # Store the settings in the dynamic config store
     get_dynamic_config_store().store(TOKEN_BUDGET_SETTINGS, settings_json)
     return {"message": "Token budget settings updated successfully."}
+
+
+@router.put("/admin/users")
+def bulk_invite_users(
+    _: User | None = Depends(current_admin_user),
+    emails: list[str] = Body(..., embed=True),
+) -> int:
+    all_emails = list(set(emails) | set(get_invited_users()))
+    return write_invited_users(all_emails)
+
+
+@router.patch("/admin/remove-invited-user")
+def remove_invited_user(
+    user_email: UserByEmail,
+    _: User | None = Depends(current_admin_user),
+) -> int:
+    users = get_invited_users()
+    remaining_users = [user for user in users if user != user_email.user_email]
+    return write_invited_users(remaining_users)
+
+
+@router.patch("/admin/block-user")
+def block_user(
+    user_email: UserByEmail,
+    _: User | None = Depends(current_admin_user),
+    db_session: Session = Depends(get_session),
+) -> None:
+    user_to_block = get_user_by_email(
+        email=user_email.user_email, db_session=db_session
+    )
+    if not user_to_block:
+        raise HTTPException(status_code=404, detail="User not found")
+
+    user_to_block.is_active = False
+    db_session.add(user_to_block)
+    db_session.commit()
+
+
+@router.patch("/admin/unblock-user")
+def unblock_user(
+    user_email: UserByEmail,
+    _: User | None = Depends(current_admin_user),
+    db_session: Session = Depends(get_session),
+) -> None:
+    user_to_unblock = get_user_by_email(
+        email=user_email.user_email, db_session=db_session
+    )
+    if not user_to_unblock:
```

Comment:
Same thing here with the logging

not a suggestion but I wonder if a user should not be returned to admin status immediately if they were blocked/disabled

Reply:
Can go either way, so maybe simpler is better?  FWIW, it's just as easy for the admin unblocking a user to adjust their privileges immediately afterwards.